lib/OpenQA/Worker/Common.pm

```diff
@@ -334,11 +334,11 @@ sub setup_websocket {
             else {
                 my $err = $tx->error;
                 $ws = undef;
-                warn "Unable to upgrade connection to WebSocket: " . $err->{code} . ". proxy_wstunnel enabled?" if defined $err;
-                if ($err->{code} eq '404') {
-                    # worker id suddenly not known anymore. Abort. If workerid
-                    # is unset we already detected that in api_call
-                    if ($workerid) {
+                if (defined $err) {
```

Comment:
The regular error handling is just printing a warning. For 404 errors in particular the worker ID is invalidated (if not already).

Not sure whether this is the correct behavior, too.